### PR TITLE
feat: add `absurd` tactic

### PR DESCRIPTION
**Pull request by fgdorais**
Handy little tactic to "smartly" apply `absurd`.

### CODE REVIEW

File: test/absurd.lean [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+-- Basic example
+/--
+error: unsolved goals
+p : Prop
+h : p
+⊢ ¬p
+-/
+#guard_msgs in
+example {p : Prop} (h : p) : False := by
```
Comment (digama0) [inline test/absurd.lean:14]: you should use goals that are not just `False` in the tests

Comment (fgdorais) [reply test/absurd.lean:14]: Any preference? Random terms? Or `sorry`?

Comment (digama0) [reply test/absurd.lean:14]: Just anything else - you can't tell from this test whether the tactic requires that the goal is `False`

File: Std/Tactic/Basic.lean [hunk 2]
```diff
@@ -69,6 +69,13 @@ macro (name := byContra) tk:"by_contra" e?:(ppSpace colGt binderIdent)? : tactic
     | refine Decidable.byContradiction fun $e => ?_
     | refine Classical.byContradiction fun $e => ?_)
 
+/--
+Given a proof `h` of `p`, `absurd h` changes the goal to `⊢ ¬ p`.
+If `p` is a negation `¬q` then the goal is changed to `⊢ q` instead.
+-/
+macro "absurd " h:term : tactic =>
+  `(tactic| first | apply absurd _ $h | apply absurd $h)
```
Comment (digama0) [inline Std/Tactic/Basic.lean:77]:
```suggestion
  `(tactic| first | refine absurd ?_ $h | refine absurd $h ?_)
```
better not to use `apply` if you know the arity

**Pull request merged**